auth-web/src/components/auth/staff/continuation-application/ContinuationApplicationTable.vue

```diff
@@ -405,8 +657,58 @@ export default defineComponent({
     }
     // Method to format dates
     const formatDate = (dateString) => {
+      if (!dateString) {
+        return '' // when no FE date, return empty string
+      }
       return moment(dateString).format('MMMM D, YYYY') // Format like "May 5, 2024"
     }
+    const formatTime = (dateString) => {
+      if (!dateString) {
+        return '' // when no FE date, return empty string
+      }
+      return moment(dateString).format('h:mm a') // Format like "12:46 pm"
+    }
+
+    // Method to check if FE date or NR date has passed
+    const expiredDate = (effectiveDate: string): boolean => {
+      return effectiveDate && !moment(effectiveDate).isAfter(moment())
+    }
+    // Method to check FE or NR days left
+    const daysLeft = (effectiveDate: string): number => {
+      const diffHours = moment(effectiveDate).diff(moment(), 'hours')
+      if (diffHours > 0 && diffHours <= 24) {
+        return 1
+      } else {
+        const diffDays = moment(effectiveDate).diff(moment(), 'days')
+        return diffDays > 0 ? diffDays : 50
+      }
+    }
+    // Method to display FE days left text
+    const feDaysLeftText = (effectiveDate: string): string => {
+      const diffHours = moment(effectiveDate).diff(moment(), 'hours')
+      if (diffHours > 0 && diffHours <= 24) {
+        return '1 day'
+      } else {
+        const diffDays = moment(effectiveDate).diff(moment(), 'days')
+        return `${diffDays} days`
+      }
+    }
+    // Method to display NR days left text
+    const nrDaysLeftText = (effectiveDate: string): string => {
+      const diffHours = moment(effectiveDate).diff(moment(), 'hours')
+      if (diffHours > 0 && diffHours <= 24) {
```

Comment:
I think this assumes that the NR will expire at the end of the day. Would it better to check if the NR expiry day is the same as today?

Reply:
Will update, thanks

Reply:
Updated

Reply:
Looks good. Resolving.